cql/DashboardLibrary.cql

```diff
@@ -362,56 +453,56 @@ define PertinentMedicationStatements:
 define PertinentMedicationRequestsSummary:
   (PertinentMedicationRequests) M
   return {
-    Type: 'Request',
-    Name: Common.ConceptText(M.medication as FHIR.CodeableConcept),
-    Start: Common.DateTimeText(M.authoredOn),
-    End:   null as String, // MedicationRequest end date not supported in R4
-    Status: M.status.value
+    name: Common.ConceptText(M.medication as FHIR.CodeableConcept),
+    value: M.status.value,
+    date: Common.DateTimeText(M.authoredOn),
+    reference: 'MedicationRequest/' + M.id
   }
-  sort by Start desc, End desc
+  sort by date desc
 
 define PertinentMedicationDispensesSummary:
   (PertinentMedicationDispenses) M
   return {
-    Type: 'Dispense',
-    Name: Common.ConceptText(M.medication as FHIR.CodeableConcept),
-    Start: Coalesce(
+    name: Common.ConceptText(M.medication as FHIR.CodeableConcept),
+    value: M.status.value,
+    date: Coalesce(
       Common.DateTimeText(M.whenHandedOver),
       Common.DateTimeText(M.whenPrepared)
     ),
-    End:   null as String, // MedicationDispense end date not supported in R4
-    Status: M.status.value
+    reference: 'MedicationDispense/' + M.id
   }
-  sort by Start desc, End desc
+  sort by date desc
 
 define PertinentMedicationAdministrationsSummary:
   (PertinentMedicationAdministrations) M
   return {
-    Type: 'Administration',
-    Name: Common.ConceptText(M.medication as FHIR.CodeableConcept),
-    Start: Common.DateTimeText((M.effective as FHIR.Period)."start"),
-    End:   Common.DateTimeText((M.effective as FHIR.Period)."end"),
-    Status: M.status.value
+    name: Common.ConceptText(M.medication as FHIR.CodeableConcept),
+    value: M.status.value,
+    date: Common.DateTimeText((M.effective as FHIR.Period)."start") +
+      ' to ' +
+      Common.DateTimeText((M.effective as FHIR.Period)."end"),
+    reference: 'MedicationAdministration/' + M.id
   }
-  sort by Start desc, End desc
+  sort by date desc
```

Comment:
@dwwinters Will sorting the date always work as we expect on a string? What about when the strings are different lengths (ex. YYYY-MM vs YYYY-MM-DD)? Or here where the word "to" is included in the string you are sorting with.

This concern was present before this PR, so I can open an issue to investigate if you think this could be something of concern.

Reply:
I created the following GitHub Issue: https://github.com/ccsm-cds-tools/ccsm-cds-with-tests/issues/27